### ISSUE TITLE
[Feature Request][output] AWS SES (simple email service)

### ISSUE THREAD

**Issue by jack1902** [airbnb/streamalert#1080]
- tl;dr I would like the ability to send alerts via AWS SES (Simple Email Service).

### Description

Whilst slack/teams acts as the IM version of alerting, having the ability to send alerts via SES, would be great for those not constantly on IM but have the ability to check emails. 

Previously, i used a lambda written in python3 to act as an output which sent the emails as i didn't want to write anything in python2. I would like to add the ability natively to SA.

Also, we should consider the ability to send HTML structured alerts instead of text based alerts via email

### Permissions (AWS)
- `ses:SendRawEmail` [Docs](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/control-user-access.html)

### Code
- use `email` package to contruct the raw mime email
- use boto3 `ses_client.send_raw_email()`to send the email

**Comment by Ryxias** [comment]
Heck yeah

I think the original reason this wasn't implemented was that it would be difficult to construct email templates. You'd have to write one, super generic (and likely not very pretty) email template that would be used for every single SES output.

With the advent of [*AlertPublishers*](https://github.com/airbnb/streamalert/pull/900), this changes a lot, because now you can define the email templates in each rule or publisher and fill out the form fields with relevant data, based upon the incoming record.

In this way, we could implement a very rudimentary `OutputDispatcher` that simply provides a default email template, but allows publishers to customize the templates via some of those "magic" variables (e.g. `@aws_ses.email_template` and `@aws_ses.email_template_vars`).

**Comment by Ryxias** [comment]
I believe the way we currently do it (not necessarily saying it's good), is we use the AWS SNS output to dispatch emails.

**Comment by jack1902** [comment]
> Heck yeah
> 
> I think the original reason this wasn't implemented was that it would be difficult to construct email templates. You'd have to write one, super generic (and likely not very pretty) email template that would be used for every single SES output.
> 
> With the advent of [_AlertPublishers_](https://github.com/airbnb/streamalert/pull/900), this changes a lot, because now you can define the email templates in each rule or publisher and fill out the form fields with relevant data, based upon the incoming record.
> 
> In this way, we could implement a very rudimentary `OutputDispatcher` that simply provides a default email template, but allows publishers to customize the templates via some of those "magic" variables (e.g. `@aws_ses.email_template` and `@aws_ses.email_template_vars`).

I used Jinja2 with a template constructed within each rule relevant publisher to make targeted alerts with potential remediation actions. This then got sent to a lambda output which sent the stuff to SES. I also attached the `alert.output_dict()` as a JSON file on all the emails I sent 

I see alot of potential for templating here where the user can construct their own templates like you mentioned

**Comment by jack1902** [comment]
Closing as work is now merged
